itest/rfq_test.go

```diff
@@ -0,0 +1,353 @@
+package itest
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightninglabs/taproot-assets/taprpc/mintrpc"
+	"github.com/lightninglabs/taproot-assets/taprpc/rfqrpc"
+	"github.com/lightningnetwork/lnd/chainreg"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/routerrpc"
+	"github.com/lightningnetwork/lnd/lntest"
+	"github.com/lightningnetwork/lnd/lntest/node"
+	"github.com/lightningnetwork/lnd/lntest/wait"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/stretchr/testify/require"
+)
+
+// testRfqHtlcIntercept tests RFQ negotiation and HTLC interception and
+// validation between three peers.
+//
+// The procedure is as follows:
+//  1. Carol sends a tap asset request for quote (buy order) to Bob.
+//  2. Bob's node accepts the quote.
+//  3. Carol uses the quote accept message to construct a lightning invoice which
+//     will pay for the quote accepted by Bob.
+//  4. Alice pays the invoice.
+//  5. Bob's node intercepts the lightning payment from Alice and validates it
+//     against the quote accepted between Bob and Carol.
+func testRfqHtlcIntercept(t *harnessTest) {
+	// Initialize a new test scenario.
+	ts := newRfqTestScenario(t)
+	defer ts.Cleanup()
```

Comment:
nit: could do `t.t.Cleanup(ts.Cleanup)` instead. Or even do that directly in `newRfqTestScenario()`.